DIPs/DIP1xxx-rn.md

```diff
@@ -0,0 +1,478 @@
+from the Python implementation [5].
+
+### Introducing the copy constructor
+
+As stated above, the fundamental problem with the postblit is the automatic blitting
+of fields which makes it impossible to type check and cannot be synchronized without
+additional costs.
+
+As an alternative, this DIP proposes the implementation of a copy constructor. The benefits
+of this approach are the following:
+
+* it is a known concept. C++ has it [3];
+* it can be typechecked as a normal constructor (since no blitting is done, the data is
+initialized as if it were in a normal constructor); this means that `const`/`immutable`/`shared`
+copy constructors will be type checked exactly as their analogous constructors
+* offers encapsulation
+
+The downside of this solution is that the user must do all the field copying by hand
+and every time a field is added to a struct, the copy constructor must be modified.
+However, this can be easily bypassed by D's introspection mechanisms. For example,
+this simple code may be used as a language idiom or library function:
+
+```D
+static foreach (i, ref field; src.tupleof)
+    this.tupleof[i] = field;
+```
+
+As shown above, the single benefit of the postblit can be easily substituted with a
+few lines of code. On the other hand, to fix the limitations of the postblit it is
+required that more complexity is added for little to no benefit. In these circumstances,
+for the sake of uniformity and consistency, replacing the postblit constructor with a
+copy constructor is the reasonable thing to do.
+
+## Description
+
+This section discusses all the technical aspects regarding the semantics
+of the copy constructor.
+
+### Syntax
+
+The following link exhibits the proposed syntax and the necessary grammar changes in order to support
+the copy construct:
+
+https://github.com/dlang/dlang.org/pull/2414
+
+The copy constructor declaration will be recognized by the parser when the tokens `@`, `implicit`, `this`
+are encountered exactly in this order. The use of a specific user defined attribute in conjunction with `this`
+benefits from the advantage of declaring the copy constructor in an expressive manner without adding any new
+keywords to the language.
+
+The argument to the copy constructor is passed by reference in order to avoid infinite recursion (passing by
+value would require a copy of the `struct` which would be made by calling the copy constructor, thus leading
+to an infinite chain of calls).
+
+The type qualifiers may be applied to the parameter of the copy constructor, but also to the function itself
+in order to facilitate the ability to describe copies between objects of different mutability levels. The type
+qualifiers are optional.
+
+### Semantics
+
+This sections discusses all the aspects regarding the semantic analysis and interaction with other
+components of the copy constructor
+
+#### Requirements
+
+1. The `structName` type needs to be identical to `typeof(this)`; an error is issued otherwise. This
```

Comment:
This took me a while to understand. It'd probably be best to say that the type of the parameter to the copy constructor needs to be identical to `typeof(this)`.